Using array of arrays causes compile error from the C compiler

mytests/testarray.nim
```nim
import std/syncio

var a: array[1, array[1, int]] = [[0]]
echo a[0][0]
```

```console
$ ./bin/nimony c -r mytests/testarray.nim 
bin/nimsem  --isSystem m nimcache/sysvq0asl.1.nif nimcache/sysvq0asl.2.nif nimcache/sysvq0asl.2.idx.nif
bin/nimsem  m nimcache/syn1lfpjv.1.nif nimcache/syn1lfpjv.2.nif nimcache/syn1lfpjv.2.idx.nif
bin/nimsem  m nimcache/tes42uk2w.1.nif nimcache/tes42uk2w.2.nif nimcache/tes42uk2w.2.idx.nif
cc -c   -Ivendor/mimalloc/include vendor/mimalloc/src/static.c -o nimcache/static.o
bin/hexer --bits:64 nimcache/tes42uk2w.2.nif
bin/nifc c --compileOnly --isMain  nimcache/tes42uk2w.c.nif
cc -c  -Imytests  nimcache/tes42uk2w.c -o nimcache/tes42uk2w.o
nimcache/tes42uk2w.c: In function ‘main’:
nimcache/tes42uk2w.c:338:45: error: subscripted value is neither array nor pointer nor vector
  338 | write_2_syn1lfpjv(stdout, a_0_tes42uk2w.a[0][0]);
      |                                             ^
```

--- a/src/nifc/typenav.nim
+++ b/src/nifc/typenav.nim
@@ -72,6 +72,14 @@ proc getTypeImpl(m: var Module; n: Cursor): Cursor =
       let a = n.firstSon
       let arrayType = getTypeImpl(m, a)
       result = arrayType
+      # array type is an alias
+      if result.kind == Symbol:
+        let d = m.defs.getOrDefault(result.symId)
+        if d.pos != 0:
+          let dd = m.src.cursorAt(d.pos)
+          if dd.stmtKind == TypeS:
+            let decl = asTypeDecl(dd)
+            result = decl.body
       inc result # (arr ...)
     of DotC:
       var a = n.firstSon

--- a/tests/nimony/sysbasics/tarrays.nim
+++ b/tests/nimony/sysbasics/tarrays.nim
@@ -1,0 +1,4 @@
+import std/syncio
+
+var a: array[1, array[1, int]] = [[0]]
+echo a[0][0]

